### ISSUE TITLE
Add support for array uniforms

### ISSUE THREAD

**Issue by Shimmen** [tsherif/picogl.js#16]
I am currently writing a simple program with PicoGL and the following piece of shader code:

```
struct Light {
    vec4 pos;
    vec4 color;
};

const int MAX_NUM_LIGHTS = 100;
layout(std140) uniform LightBlock
{
    Light lights[MAX_NUM_LIGHTS];
    int numLights;
};

```

The current system in PicoGL for both uniforms and uniform blocks can only handle non-array types if I understand the code correctly, which makes this difficult to do.

It would be very useful if you could specify uniform block types with some number of elements too. For the above example you could maybe do something like:

```
var uniformBuffer = app.createUniformBuffer([
    {type: PicoGL.FLOAT_VEC4, numElements: 100 * 2},
    PicoGL.INT
])
``` 

There are of course other options, like using a texture or similar for storing the data, but I feel like this should be possible to do in PicoGL for feature coverage.

**Comment by Shimmen** [comment]
I have also slightly considered possible implementations of this.

For simple non-block uniforms it mostly already works. The problem is with the caching since it probably doesn't make sense to cache a huge array. So when setting up the uniforms for a program you could check the `size` parameter of the uniform info object and if it indicates that it's an array you for example create a new `UintArrayUniform(gl, uniformHandle, count)` or similar.

For uniforms blocks it doesn't have to be much more complicated but you will have to increment the size of the buffers etc. when setting up the buffer storage.

I could work on a PR if it sounds good.

**Comment by tsherif** [comment]
@Shimmen yes, this totally makes sense. Sorry I didn't reply sooner. Apparently, I had notifications for this repo turned off. Will look at your PR shortly.

**Comment by tsherif** [comment]
@Shimmen For the uniform blocks, I'm a little concerned since you can work with uniform arrays; you just treat each element as separate, e.g. For your example, it would look something like this:

```JavaScript
var layout = new Array(201);
for (var i = 0; i < 200; ++i) {
    layout[i] = PicoGL.FLOAT_VEC4;
}
layout[200] = PicoGL.INT;
var uniformBuffer = app.createUniformBuffer(layout);
```

**Comment by Shimmen** [comment]
Oh, of course that works for the uniform blocks! It never occurred to me before you said it. But it's actually quite convenient to set it up with a loop, so never mind that part.

**Comment by tsherif** [comment]
Resolved by #17
